### ISSUE TITLE
Support for (All) Linked Services for Azure Synapse

### ISSUE THREAD

**Issue by gro1m** [hashicorp/terraform-provider-azurerm#10979]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Manual steps to create a linked service for Azure Synapse is via:
1. Open the Azure Synapse Studio and go to the Manage tab.
2. Under External connections, select Linked services.
3. To add a linked service, select New.

It would be nice if one could create an Azure Synapse workspace that comes with the possibility of defining linked services as in the list that pops up when doing the manual steps 1-3 above.

### New or Affected Resource(s)

* azurerm_synapse_(workspace)_linked_service(s)

### Potential Terraform Configuration

I am not sure about this, as I do not know if it is really feasible given that there is a long list of services that can be linked to. Would be nice to have all database services, but there are a lot.

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by w0ut0** [comment]
Is it a lot of work to migrate/duplicate the existing (data factory) linked services so they support Synapse?

- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_blob_storage
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_databricks
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_databricks
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_function
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_sql_database
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_table_storage
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_cosmosdb
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_data_lake_storage_gen2
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_key_vault
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_mysql
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_postgresql
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_sftp
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_snowflake
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_sql_server
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_synapse
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_web

**Comment by little-goat-LCY** [comment]
> Is it feasible to migrate/duplicate the existing (data factory) linked services so they support Synapse?
> 
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_blob_storage
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_databricks
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_databricks
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_function
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_sql_database
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_azure_table_storage
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_cosmosdb
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_data_lake_storage_gen2
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_key_vault
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_mysql
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_postgresql
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_sftp
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_snowflake
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_sql_server
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_synapse
> * https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/data_factory_linked_service_web

I would also be interested in the possibility? Did anyone find a solution?

**Comment by owenfarrell** [comment]
I believe this was fixed in #13204

**Comment by little-goat-LCY** [comment]
> I believe this was fixed in #13204

No, I think it wasn't a bug fix. It was rather a version update. The lastest version of azurerm lib does already include this function :) But thx though!

**Comment by gro1m** [comment]
Fixed by @13204.
